### PR TITLE
Bring in launch binary and console runner library from common

### PR DESCRIPTION
**Pull request by dmitrii-ubskii**
## Usage and product changes

We move the `binary` package and `console-runner` into this repository from typedb-common. `typedb-console-runner` is deployed to maven such that we can safely depend on it from other repos without creating Bazel dependency cycles.

### CODE REVIEW

File: dependencies/vaticle/repositories.bzl [hunk 1]
```diff
@@ -20,15 +20,15 @@ load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
 def vaticle_dependencies():
     git_repository(
         name = "vaticle_dependencies",
-        remote = "https://github.com/vaticle/dependencies",
-        commit = "371a67999d92b4d0833cd5e016ec01c6278e72c4",  # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_dependencies
+        remote = "https://github.com/dmitrii-ubskii/dependencies",
+        commit = "27232484e387154a60c27e3c9d76698a677e5c30", # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_dependencies
```
Comment (dmitrii-ubskii) [inline dependencies/vaticle/repositories.bzl:24]: Note.

File: dependencies/vaticle/repositories.bzl [hunk 2]
```diff
@@ -20,15 +20,15 @@ load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
...
     )
 
 def vaticle_typedb_common():
     git_repository(
         name = "vaticle_typedb_common",
-        remote = "https://github.com/vaticle/typedb-common",
-        tag = "2.26.0",  # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_typedb_common
+        remote = "https://github.com/dmitrii-ubskii/typedb-common",
+        commit = "7fd4316e7fd4c3ba9ce94f2ea73f12c82057930b",  # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_typedb_common
```
Comment (dmitrii-ubskii) [inline dependencies/vaticle/repositories.bzl:31]: Note

File: test/assembly/BUILD [hunk 3]
```diff
@@ -18,12 +18,17 @@
...
 
 typedb_java_test(
     name = "test-assembly-native",
     test_class = "com.vaticle.typedb.console.test.assembly.AssemblyTest",
     srcs = ["AssemblyTest.java"],
+    deps = [
+        "//test/runner:typedb-console-runner",
+        "@maven//:com_vaticle_typedb_typedb_runner",
+    ],
+    classpath_resources = ["//test/runner:logback"],
```
Comment (flyingsilverfin) [inline test/assembly/BUILD:31]: Why do we have 'logback' here and another in 'dependencies' as well?

File: dependencies/maven/artifacts.bzl [hunk 4]
```diff
@@ -26,3 +26,7 @@ artifacts = [
   "org.hamcrest:hamcrest-library",
   "org.slf4j:slf4j-api"
 ]
+
+internal_artifacts = {
```
Comment (flyingsilverfin) [inline dependencies/maven/artifacts.bzl:30]: 'vaticle_artifacts'

**Review by flyingsilverfin — APPROVED** [review]

**Pull request merged**